ambry-store/src/main/java/com/github/ambry/store/LogSegment.java

```diff
@@ -83,6 +84,7 @@
           StoreErrorCodes.File_Not_Found);
     }
     this.file = file;
+    this.diskMountPath = parseDiskMountPath(file);
```

Comment:
Can we get this `diskMountPath` from `BlobStore`?

Reply:
If we want to get diskMountPath from BlobStore, we need to get it from `dataDir`.
This is not necessary, because in `Log.java`, we can directly get `diskMountPath` from parsing `dataDir`  and then pass `diskMountPath` to LogSegment.java. 
However, complexity wise, I didn't feel it's better than current solution. 

@cgtz  do you have any suggestions?

Reply:
What if we had the `BlobStore` pass down the metrics themselves. You could even make a small class to hold the three metrics together:
```
class DiskMetrics {
  Histogram diskRead, diskWrite;
  Meter copyRate
}
```
This would avoid any need to use maps, I feel, and is cleaner IMO, at the expense of more boilerplate constructor changes.

1. DiskManager constructs a DiskMetrics for the disk that it manages. It already knows it represents a single disk.
2. BlobStore takes in DiskMetrics in its constructor
3. LogSegment and BlobStoreCompactor take in DiskMetrics objects. They no longer need to worry about the mount path or hashmap lookups.